cgsmiles/sample.py

```diff
@@ -202,9 +212,10 @@ def __init__(self,
             if not key[-1].isdigit():
                 key += '1'
             self.fragment_reactivities[key] = _set_bond_order_defaults(probs)
-        self.terminal_reactivities = _set_bond_order_defaults(terminal_reactivities)
-        self.branch_term_probs = branch_term_probs
+
         self.all_atom = all_atom
+        # make sure to get bond order defaults
+        self.terminal_bonds = _set_bond_order_defaults(terminal_bonds)
```

Comment:
Normalize `terminal_bonds` to be a dict here.

Reply:
but that makes no sense what would be the corresponding key:value?

Reply:
My brain is mushy, what does the dict look like if user provided?

Reply:
terminal_bonds is a list with terminal bonding descriptors not a dict

Reply:
... but `_set_bond_order_defaults` can also deal with dicts

Reply:
yes the dict are the fragment_reactivities:

{'bond_1': {bond1: 0.0, bond2: 1.0},
 'bond_2': {bond2: 0.0 bond1: 1.0}}

Reply:
set bond_order just adds a 1 to all bonding descriptors if they don't end in a digit and I needed it to work for a list and double dict. Hence that private function